src/display.rs

```diff
@@ -251,28 +255,50 @@ fn get_long_output(
     meta: &Meta,
     colors: &Colors,
     icons: &Icons,
-    flags: Flags,
+    flags: &Flags,
     padding_rules: PaddingRules,
 ) -> String {
-    let strings: &[ANSIString] = &[
-        meta.file_type.render(colors),
-        meta.permissions.render(colors),
-        ANSIString::from(" "),
-        meta.owner.render_user(colors, padding_rules.user),
-        ANSIString::from(" "),
-        meta.owner.render_group(colors, padding_rules.group),
-        ANSIString::from(" "),
-        meta.size
-            .render(colors, padding_rules.size.0, padding_rules.size.1, flags),
-        ANSIString::from(" "),
-        meta.date.render(colors, padding_rules.date, flags),
-        ANSIString::from(" "),
-        meta.name.render(colors, icons),
-        meta.indicator.render(flags),
-        meta.symlink.render(colors),
-    ];
+    let mut strings: Vec<ANSIString> = Vec::new();
+    for block in flags.blocks.iter() {
+        match block {
+            Block::Permission => {
+                strings.push(meta.file_type.render(colors));
+                strings.push(meta.permissions.render(colors));
+            }
+            Block::User => strings.push(meta.owner.render_user(colors, padding_rules.user)),
+            Block::Group => strings.push(meta.owner.render_group(colors, padding_rules.group)),
+            Block::Size => strings.push(meta.size.render(
+                colors,
+                padding_rules.size.0,
+                padding_rules.size.1,
+                &flags,
+            )),
+            Block::Date => strings.push(meta.date.render(colors, padding_rules.date, &flags)),
+            Block::Name => strings.push(meta.name.render(colors, icons, Some(padding_rules.name))),
+            Block::NameWithSymlink => match meta.symlink.symlink_string() {
+                Some(_) => {
+                    strings.push(meta.name.render(colors, icons, None));
+                    strings.push(meta.indicator.render(&flags));
+                    strings.push(meta.symlink.render(
+                        colors,
+                        Some(padding_rules.name_with_symlink - meta.name.name_string(icons).len()),
+                    ));
+                }
+                None => {
+                    strings.push(meta.name.render(
+                        colors,
+                        icons,
+                        Some(padding_rules.name_with_symlink + 3),
+                    ));
+                    strings.push(meta.indicator.render(&flags));
+                    strings.push(meta.symlink.render(colors, None));
```

Comment:
I don't think you need to add it in this condition. It should always return an empty string.

Reply:
I am not sure what you are referring to.

Reply:
I think I figured out what you where talking about. I have made it to not do padding inside.